### ISSUE TITLE
ProtoScenario.preHookMessages not visible in Json-Report

### ISSUE THREAD

**Issue by r-vanooyen** [getgauge-contrib/json-report#18]
Hello,

As a report analyst i want to see the preHookMessages of the scenario in the json report.
In a html-report the preHookMessages of all stages are visible, e.g. [here](https://github.com/getgauge/html-report/blob/decc4783a592163a39ecd699a9b0f26d8d80fa01/generator/generate.go#L157) for the scenario stage.

Currently only messages of the step stage are shown in the json report

**Comment by r-vanooyen** [comment]
This worked on my maschine:

```Index: generate.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- generate.go	(revision 3820f56cb30d14b1b9225cff65375479a70cde6c)
+++ generate.go	(revision 3a7d4c4b924463540884a9e530abdca0d100cc0a)
@@ -67,17 +67,18 @@
 }
 
 type scenario struct {
-	Heading                   string       `json:"scenarioHeading"`
-	Tags                      []string     `json:"tags"`
-	ExecutionTime             int64        `json:"executionTime"`
-	ExecutionStatus           status       `json:"executionStatus"`
-	Contexts                  []item       `json:"contexts"`
-	Teardowns                 []item       `json:"teardowns"`
-	Items                     []item       `json:"items"`
-	BeforeScenarioHookFailure *hookFailure `json:"beforeScenarioHookFailure"`
-	AfterScenarioHookFailure  *hookFailure `json:"afterScenarioHookFailure"`
-	SkipErrors                []string     `json:"skipErrors"`
-	TableRowIndex             int          `json:"tableRowIndex"`
+	Heading                    string       `json:"scenarioHeading"`
+	Tags                       []string     `json:"tags"`
+	ExecutionTime              int64        `json:"executionTime"`
+	ExecutionStatus            status       `json:"executionStatus"`
+	Contexts                   []item       `json:"contexts"`
+	Teardowns                  []item       `json:"teardowns"`
+	Items                      []item       `json:"items"`
+	BeforeScenarioHookMessages []string     `json:"beforeScenarioHookMessages"`
+	BeforeScenarioHookFailure  *hookFailure `json:"beforeScenarioHookFailure"`
+	AfterScenarioHookFailure   *hookFailure `json:"afterScenarioHookFailure"`
+	SkipErrors                 []string     `json:"skipErrors"`
+	TableRowIndex              int          `json:"tableRowIndex"`
 }
 
 type step struct {
@@ -200,18 +201,22 @@
 
 func toScenario(protoSce *gauge_messages.ProtoScenario, tableRowIndex int) scenario {
 	sce := scenario{
-		Heading:                   protoSce.GetScenarioHeading(),
-		ExecutionTime:             protoSce.GetExecutionTime(),
-		Tags:                      make([]string, 0),
-		ExecutionStatus:           getScenarioStatus(protoSce),
-		Contexts:                  make([]item, 0),
-		Items:                     make([]item, 0),
-		Teardowns:                 make([]item, 0),
-		BeforeScenarioHookFailure: toHookFailure(protoSce.GetPreHookFailure()),
-		AfterScenarioHookFailure:  toHookFailure(protoSce.GetPostHookFailure()),
-		TableRowIndex:             tableRowIndex,
-		SkipErrors:                make([]string, 0),
+		Heading:                    protoSce.GetScenarioHeading(),
+		ExecutionTime:              protoSce.GetExecutionTime(),
+		Tags:                       make([]string, 0),
+		ExecutionStatus:            getScenarioStatus(protoSce),
+		Contexts:                   make([]item, 0),
+		Items:                      make([]item, 0),
+		Teardowns:                  make([]item, 0),
+		BeforeScenarioHookMessages: make([]string, 0),
+		BeforeScenarioHookFailure:  toHookFailure(protoSce.GetPreHookFailure()),
+		AfterScenarioHookFailure:   toHookFailure(protoSce.GetPostHookFailure()),
+		TableRowIndex:              tableRowIndex,
+		SkipErrors:                 make([]string, 0),
 	}
+	if protoSce.GetPreHookMessages() != nil {
+		sce.BeforeScenarioHookMessages = protoSce.GetPreHookMessages()
+	}
 	if protoSce.GetSkipErrors() != nil {
 		sce.SkipErrors = protoSce.GetSkipErrors()
 	}
```

**Comment by sriv** [comment]
Would you be interested in sending a Pull Request @r-vanooyen ? Given you've analyzed and figured out a solution?

**Comment by sriv** [comment]
closed via #21
